CTFd/api/v1/challenges.py

```diff
@@ -182,22 +182,24 @@ def get(self, query_args):
         admin_view = is_admin() and request.args.get("view") == "admin"
 
         solve_counts, user_solves = dict(), set()
+        # Build a query for to show challenge solve information. We only
+        # give an admin view if the request argument has been provided.
+        #
+        # NOTE: This is different behaviour to the challenge detail
+        # endpoint which only needs the current user to be an admin rather
+        # than also also having to provide `view=admin` as a query arg.
+        solves_q = _build_solves_query(admin_view=admin_view)
+        # Aggregate the query results into the hashes defined at the top of
+        # this block for later use
+        for chal_id, solve_count, solved_by_user in solves_q:
+            solve_counts[chal_id] = solve_count
+            if solved_by_user:
+                user_solves.add(chal_id)
         if scores_visible() and accounts_visible():
             solve_count_dfl = 0
-            # Build a query for to show challenge solve information. We only
-            # give an admin view if the request argument has been provided.
-            #
-            # NOTE: This is different behaviour to the challenge detail
-            # endpoint which only needs the current user to be an admin rather
-            # than also also having to provide `view=admin` as a query arg.
-            solves_q = _build_solves_query(admin_view=admin_view)
-            # Aggregate the query results into the hashes defined at the top of
-            # this block for later use
-            for chal_id, solve_count, solved_by_user in solves_q:
-                solve_counts[chal_id] = solve_count
-                if solved_by_user:
-                    user_solves.add(chal_id)
         else:
+            # Empty out the solves_count if we're hiding scores/accounts
+            solve_counts = {}
```

Comment:
This seems a bit unfortunately wasteful to do :( I'd guess that having scores off isn't that common though and since we need to find any solve for every challenge ID, making a simpler query to just get a value for `solved_by_me` seems like it's not worth the effort.

seems legit!

Reply:
Yeah I think double query isn't worth it but you're right it's not an extremely common configuration. 

I think I'll just see if it ends up causing problems and I'll optimize further when it gets there.